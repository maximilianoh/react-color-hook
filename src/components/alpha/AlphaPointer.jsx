import React from 'react';
import reactCSS from 'reactcss';
import PropTypes from 'prop-types';

const AlphaPointer = ({ direction }) => {
  const styles = reactCSS({
    default: {
      picker: {
        width: '18px',
        height: '18px',
        borderRadius: '50%',
        transform: 'translate(-9px, -1px)',
        backgroundColor: 'rgb(248, 248, 248)',
        boxShadow: '0 1px 4px 0 rgba(0, 0, 0, 0.37)',
      },
    },
    vertical: {
      picker: {
        transform: 'translate(-2px, -9px)',
      },
    },
  }, { vertical: direction === 'vertical' });

  return (
    <div style={styles.picker} />
  );
};

AlphaPointer.defaultProps = {
  direction: 'horizontal',
};

AlphaPointer.propTypes = {
  direction: PropTypes.string,
};
export default AlphaPointer;
