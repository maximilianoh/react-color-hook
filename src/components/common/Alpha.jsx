import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import calculateChange from '../../helpers/alpha';
import { alphaStyle } from './commonStyles';
import Checkboard from './Checkboard';

const Alpha = (props) => {
  const inputRef = useRef(null);
  const handleChange = (e) => {
    const change = calculateChange(e, props.hsl, props.direction, props.a, inputRef.current);
    if (change && typeof props.onChange === 'function') props.onChange(change, e);
  };
  const handleMouseUp = () => {
    window.removeEventListener('mousemove', handleChange);
    window.removeEventListener('mouseup', handleMouseUp);
  };

  const handleMouseDown = (e) => {
    handleChange(e);
    window.addEventListener('mousemove', handleChange);
    window.addEventListener('mouseup', handleMouseUp);
  };


  useEffect(() => { handleMouseUp(); }, []);

  const { rgb, renderers, pointer } = props;
  const styles = alphaStyle(props, rgb);
  return (
    <div style={styles.alpha}>
      <div style={styles.checkboard}>
        <Checkboard renderers={renderers} />
      </div>
      <div style={styles.gradient} />
      <div
        style={styles.container}
        ref={inputRef}
        role="button"
        tabIndex={0}
        onMouseDown={handleMouseDown}
        onTouchMove={handleChange}
        onTouchStart={handleChange}
      >
        <div style={styles.pointer}>
          {pointer ? (
            <props.pointer {...props} />
          ) : (
            <div style={styles.slider} />
          )}
        </div>
      </div>
    </div>
  );
};

export default Alpha;


Alpha.defaultProps = {
  a: null,
  renderers: {},
  direction: 'horizontal',
  pointer: null,
};
Alpha.propTypes = {
  hsl: PropTypes.shape({}).isRequired,
  direction: PropTypes.string,
  a: PropTypes.number,
  rgb: PropTypes.shape({}).isRequired,
  renderers: PropTypes.shape({}),
  pointer: PropTypes.func,
  onChange: PropTypes.func.isRequired,
};
