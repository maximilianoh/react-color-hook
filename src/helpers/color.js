import each from 'lodash/each';
import chroma from 'chroma-js';

const isNumber = (value) => !Number.isNaN(parseFloat(value));

export const simpleCheckForValidColor = (data) => {
  const keysToCheck = ['r', 'g', 'b', 'a', 'h', 's', 'l', 'v'];
  let checked = 0;
  let passed = 0;
  each(keysToCheck, (letter) => {
    if (data[letter]) {
      checked += 1;
      if (isNumber(data[letter])) {
        passed += 1;
      }
      if (letter === 's' || letter === 'l') {
        const percentPatt = /^\d+%$/;
        if (percentPatt.test(data[letter])) {
          passed += 1;
        }
      }
    }
  });
  return (checked === passed) ? data : false;
};

export const chromaValidation = (data) => {
  let validation = false;
  try {
    if (data.hex) chroma(data.hex);
    else chroma(data);
    validation = true;
  } catch (error) {
    validation = false;
  }
  return validation;
};

export const isValidToChange = (data, colors) => {
  const key = data.source;
  if (key === 'rgb') {
    const validation = Number(data.r) === colors.r
    && Number(data.g) === colors.g && Number(data.g) === colors.g;
    return validation;
  }
  if (key === 'hsv') {
    const s = data.s > 1 ? data.s / 100 : data.s;
    const v = data.v > 1 ? data.v / 100 : data.v;
    const validation = Number(data.h) >= 0 && Number(data.h) <= 360
    && Number(s) >= 0 && Number(s) <= 1
    && Number(v) >= 0 && Number(v) <= 1;
    return validation;
  }
  if (key === 'hsl') {
    const s = data.l > 1 ? data.l / 100 : data.l;
    const l = data.l > 1 ? data.l / 100 : data.l;
    try {
      chroma.hsl(data.h, s, l);
      return true;
    } catch (error) {
      return false;
    }
  } else { // hex
    return data.hex.toUpperCase() === colors.toUpperCase();
  }
};

const valueNaN = (h) => (!isNumber(h) ? 0 : h);
const hslaListToObject = (list) => ({
  h: valueNaN(list[0]), s: valueNaN(list[1]), l: valueNaN(list[2]), a: valueNaN(list[3]),
});
const rgbaListToObject = (list) => ({
  r: list[0], g: list[1], b: list[2], a: list[3],
});
const hsvListToObject = (list) => ({
  h: valueNaN(list[0]),
  s: valueNaN(list[1]),
  v: valueNaN(list[2]),
});

export const hsvParse = (color, data) => {
  const s = data.s > 1 ? data.s / 100 : data.s;
  const v = data.v > 1 ? data.v / 100 : data.v;
  const c = chroma.hsv(Number(data.h), Number(s), Number(v));
  const result = {
    ...color,
    hsv: {
      h: Number(data.h),
      s: Number(s),
      v: Number(v),
    },
    hsl: hslaListToObject(c.hsl()),
    rgb: rgbaListToObject(c.rgba()),
    hex: c.hex(),
  };
  return result;
};

export const toState = (data, oldHue) => {
  const dataValidation = data || '#000000';
  let transparent = false;
  let colorChroma;
  let a = data.a ? data.a : 1;
  if (data.hex && (data.hex === 'transparent' || data === 'transparent')) {
    colorChroma = chroma('#000000').alpha(0);
    a = 0;
    transparent = true;
  } else if (data.hex) colorChroma = chroma(data.hex);
  else if (data.h && data.s && data.l) {
    const s = data.s > 1 ? data.s / 100 : data.s;
    const l = data.l > 1 ? data.l / 100 : data.l;
    colorChroma = chroma.hsl(data.h, s, l);
  } else {
    try {
      colorChroma = chroma(dataValidation);
    } catch (error) {
      colorChroma = chroma('#000000');
    }
  }
  const hsl = hslaListToObject(colorChroma.alpha(a).hsl());
  const hsv = hsvListToObject(colorChroma.hsv());
  const rgba = colorChroma.alpha(a).rgba();
  const rgb = rgbaListToObject(rgba);
  const hex = colorChroma.alpha(1).hex();

  return {
    hsl,
    hex: transparent ? 'transparent' : hex,
    rgb,
    hsv,
    oldHue: data.h || oldHue || hsl.h,
    source: data.source,
  };
};

export const isValidHex = (hex) => {
  // disable hex4 and hex8
  // const lh = (String(hex).charAt(0) === '#') ? 1 : 0;
  // const result = hex.length !== (4 + lh) && hex.length < (7 + lh) && chroma.valid(hex);
  // const hasHas = hex.indexOf('#') !== 0 ? '#' : '';

  let validation = false;
  try {
    validation = chroma.valid(hex) && typeof hex === 'string';
  } catch (error) {
    validation = false;
  }
  return validation;
};

export const getContrastingColor = (data) => {
  if (!data) {
    return '#fff';
  }
  let c = data;
  if (data === 'transparent') c = { hex: 'transparent' };
  const col = toState(c);
  if (col.hex === 'transparent') {
    return 'rgba(0,0,0,0.4)';
  }
  const yiq = ((col.rgb.r * 299) + (col.rgb.g * 587) + (col.rgb.b * 114)) / 1000;
  return (yiq >= 128) ? '#000' : '#fff';
};
