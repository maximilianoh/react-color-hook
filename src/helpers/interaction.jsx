import React, { useState } from 'react';

const handleFocus = (Component, Span = 'span') => {
  const Focus = (props) => {
    const [focus, setFocus] = useState(false);

    const handlerFocus = () => setFocus(true);

    const handleBlur = () => setFocus(false);

    return (
      <Span onFocus={handlerFocus} onBlur={handleBlur}>
        <Component {...props} {...focus} />
      </Span>
    );
  };
  return Focus;
};
export default handleFocus;
