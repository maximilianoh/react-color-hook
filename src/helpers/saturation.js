const isNumber = (value) => !Number.isNaN(parseFloat(value));
const calculateChange = (e, hsl, container) => {
  const containerWidth = container.clientWidth;
  const containerHeight = container.clientHeight;
  const x = typeof e.pageX === 'number' ? e.pageX : e.touches[0].pageX;
  const y = typeof e.pageY === 'number' ? e.pageY : e.touches[0].pageY;
  let left = x - (container.getBoundingClientRect().left + window.pageXOffset);
  let top = y - (container.getBoundingClientRect().top + window.pageYOffset);

  if (left < 0) {
    left = 0;
  } else if (left > containerWidth) {
    left = containerWidth;
  }
  if (top < 0) {
    top = 0;
  } else if (top > containerHeight) {
    top = containerHeight;
  }

  const saturation = left / containerWidth;
  const bright = 1 - (top / containerHeight);
  return {
    h: hsl.h,
    s: isNumber(saturation) ? saturation : 0,
    v: isNumber(bright) ? bright : 0,
    a: hsl.a,
    source: 'hsv',
  };
};

export default calculateChange;
