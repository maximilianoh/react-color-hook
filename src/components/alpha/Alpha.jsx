import React from 'react';
import reactCSS from 'reactcss';
import PropTypes from 'prop-types';
import Alpha from '../common/Alpha';
import ColorWrap from '../common/ColorWrap';
import AlphaPointer from './AlphaPointer';

export const AlphaPicker = ({
  rgb, hsl, width, height, onChange, direction, style,
  renderers, pointer, className = '',
}) => {
  const styles = reactCSS({
    default: {
      picker: {
        position: 'relative',
        width: `${width}px`,
        height: `${height}px`,
      },
      alpha: {
        radius: '2px',
        style,
      },
    },
  });
  debugger; // eslint-disable-line no-debugger
  return (
    <div style={styles.picker} className={`alpha-picker ${className}`}>
      <Alpha
        {...styles.alpha}
        rgb={rgb}
        hsl={hsl}
        pointer={pointer}
        renderers={renderers}
        onChange={onChange}
        direction={direction}
      />
    </div>
  );
};

AlphaPicker.defaultProps = {
  width: 316,
  height: 16,
  direction: 'horizontal',
  pointer: AlphaPointer,
  renderers: {},
  className: '',
  style: {},
};


AlphaPicker.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  direction: PropTypes.string,
  pointer: PropTypes.func,
  rgb: PropTypes.shape({}).isRequired,
  hsl: PropTypes.shape({}).isRequired,
  onChange: PropTypes.func.isRequired,
  style: PropTypes.shape({}),
  renderers: PropTypes.shape({}),
  className: PropTypes.string,

};


export default ColorWrap(AlphaPicker);
