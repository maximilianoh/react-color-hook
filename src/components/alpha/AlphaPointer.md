Alpha
```js
import AlphaPointer from './AlphaPointer.jsx';
<AlphaPointer />
```

