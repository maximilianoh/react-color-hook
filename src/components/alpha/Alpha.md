Alpha
```js
import Alpha from './Alpha.jsx';
<Alpha />
```


Alpha Color Red
```js
import Alpha from './Alpha.jsx';
import { red } from '../../helpers/color';

<Alpha {...red} color={red.hex} />
```


Alpha renders vertically
```js
import Alpha from './Alpha.jsx';
import { red } from '../../helpers/color';

<Alpha {...red} width={20} height={100} direction="vertical" />
```

Alpha onChange events
```js
import Alpha from './Alpha.jsx';
import { red } from '../../helpers/color';

const changeSpy = (prop) => { console.log(prop.hsl.a)};

<div>
    Change and see Console
    <Alpha {...red} onChange={changeSpy} />
</div>
```
