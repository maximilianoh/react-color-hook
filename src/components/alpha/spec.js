import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { simpleCheckForValidColor, red } from '../../helpers/color';
import Alpha from './Alpha';
import AlphaPointer from './AlphaPointer';
import CanvasRenderingContext2DEvent from "jest-canvas-mock";

test('Alpha renders correctly', () => {
  const tree = renderer.create(
    <Alpha {...red} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

 test('Alpha renders on server correctly', () => {
   const tree = renderer.create(
     <Alpha renderers={ CanvasRenderingContext2DEvent } { ...red } />
   ).toJSON()
   expect(tree).toMatchSnapshot()
 })

test('Alpha renders vertically', () => {
  const tree = renderer.create(
    <Alpha {...red} width={20} height={200} direction="vertical" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

test('AlphaPointer renders correctly', () => {
  const tree = renderer.create(
    <AlphaPointer />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
})


test('Alpha onChange events correctly', () => {

  const changeSpy = jest.fn((data) => {
    expect(simpleCheckForValidColor(data)).toBeTruthy();
  })
  render(<Alpha {...red} width={20} height={200} onChange={changeSpy} />);
  expect(changeSpy).toHaveBeenCalledTimes(0);

  const event = new MouseEvent('mousedown', { bubbles: true });

  Object.defineProperty(event, 'pageX', {get: () => 100});
  Object.defineProperty(event, 'pageY', {get: () => 10});

  const root = document.querySelector("div");
  const alphaPointer = root.children[0].children[0].children[2];
  fireEvent(alphaPointer, event);
  expect(changeSpy).toHaveBeenCalled();
})


test('Alpha class check', () => {
  render(<Alpha {...red} width={20} height={200} direction="vertical" className="classExample" />);
  expect(document.querySelector(".classExample")!==null).toBe(true);
})