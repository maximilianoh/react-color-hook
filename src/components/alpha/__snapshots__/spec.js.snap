// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alpha renders correctly 1`] = `
<div
  className="alpha-picker "
  style={
    Object {
      "height": "16px",
      "position": "relative",
      "width": "316px",
    }
  }
>
  <div
    style={
      Object {
        "MozBorderRadius": "2px",
        "OBorderRadius": "2px",
        "WebkitBorderRadius": "2px",
        "borderRadius": "2px",
        "bottom": "0px",
        "left": "0px",
        "msBorderRadius": "2px",
        "position": "absolute",
        "right": "0px",
        "top": "0px",
      }
    }
  >
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "OBorderRadius": "2px",
          "WebkitBorderRadius": "2px",
          "borderRadius": "2px",
          "bottom": "0px",
          "left": "0px",
          "msBorderRadius": "2px",
          "overflow": "hidden",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <div
        style={
          Object {
            "MozBorderRadius": "",
            "MozBoxShadow": "",
            "OBorderRadius": "",
            "OBoxShadow": "",
            "WebkitBorderRadius": "",
            "WebkitBoxShadow": "",
            "background": "url(data:image/png;base64,00) center left",
            "borderRadius": "",
            "bottom": "0px",
            "boxShadow": "",
            "left": "0px",
            "msBorderRadius": "",
            "msBoxShadow": "",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "MozBoxShadow": undefined,
          "OBorderRadius": "2px",
          "OBoxShadow": undefined,
          "WebkitBorderRadius": "2px",
          "WebkitBoxShadow": undefined,
          "background": "linear-gradient(to right, rgba(34,25,77, 0) 0%,
             rgba(34,25,77, 1) 100%)",
          "borderRadius": "2px",
          "bottom": "0px",
          "boxShadow": undefined,
          "left": "0px",
          "msBorderRadius": "2px",
          "msBoxShadow": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "height": "100%",
          "margin": "0 3px",
          "outline": "none",
          "position": "relative",
        }
      }
      tabIndex={0}
    >
      <div
        style={
          Object {
            "left": "100%",
            "position": "absolute",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "50%",
              "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "MozTransform": "translate(-9px, -1px)",
              "OBorderRadius": "50%",
              "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "OTransform": "translate(-9px, -1px)",
              "WebkitBorderRadius": "50%",
              "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "WebkitTransform": "translate(-9px, -1px)",
              "backgroundColor": "rgb(248, 248, 248)",
              "borderRadius": "50%",
              "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "height": "18px",
              "msBorderRadius": "50%",
              "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "msTransform": "translate(-9px, -1px)",
              "transform": "translate(-9px, -1px)",
              "width": "18px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Alpha renders on server correctly 1`] = `
<div
  className="alpha-picker "
  style={
    Object {
      "height": "16px",
      "position": "relative",
      "width": "316px",
    }
  }
>
  <div
    style={
      Object {
        "MozBorderRadius": "2px",
        "OBorderRadius": "2px",
        "WebkitBorderRadius": "2px",
        "borderRadius": "2px",
        "bottom": "0px",
        "left": "0px",
        "msBorderRadius": "2px",
        "position": "absolute",
        "right": "0px",
        "top": "0px",
      }
    }
  >
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "OBorderRadius": "2px",
          "WebkitBorderRadius": "2px",
          "borderRadius": "2px",
          "bottom": "0px",
          "left": "0px",
          "msBorderRadius": "2px",
          "overflow": "hidden",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <div
        style={
          Object {
            "MozBorderRadius": "",
            "MozBoxShadow": "",
            "OBorderRadius": "",
            "OBoxShadow": "",
            "WebkitBorderRadius": "",
            "WebkitBoxShadow": "",
            "background": "url(data:image/png;base64,00) center left",
            "borderRadius": "",
            "bottom": "0px",
            "boxShadow": "",
            "left": "0px",
            "msBorderRadius": "",
            "msBoxShadow": "",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "MozBoxShadow": undefined,
          "OBorderRadius": "2px",
          "OBoxShadow": undefined,
          "WebkitBorderRadius": "2px",
          "WebkitBoxShadow": undefined,
          "background": "linear-gradient(to right, rgba(255,0,0, 0) 0%,
             rgba(255,0,0, 1) 100%)",
          "borderRadius": "2px",
          "bottom": "0px",
          "boxShadow": undefined,
          "left": "0px",
          "msBorderRadius": "2px",
          "msBoxShadow": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "height": "100%",
          "margin": "0 3px",
          "outline": "none",
          "position": "relative",
        }
      }
      tabIndex={0}
    >
      <div
        style={
          Object {
            "left": "100%",
            "position": "absolute",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "50%",
              "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "MozTransform": "translate(-9px, -1px)",
              "OBorderRadius": "50%",
              "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "OTransform": "translate(-9px, -1px)",
              "WebkitBorderRadius": "50%",
              "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "WebkitTransform": "translate(-9px, -1px)",
              "backgroundColor": "rgb(248, 248, 248)",
              "borderRadius": "50%",
              "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "height": "18px",
              "msBorderRadius": "50%",
              "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "msTransform": "translate(-9px, -1px)",
              "transform": "translate(-9px, -1px)",
              "width": "18px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Alpha renders vertically 1`] = `
<div
  className="alpha-picker "
  style={
    Object {
      "height": "200px",
      "position": "relative",
      "width": "20px",
    }
  }
>
  <div
    style={
      Object {
        "MozBorderRadius": "2px",
        "OBorderRadius": "2px",
        "WebkitBorderRadius": "2px",
        "borderRadius": "2px",
        "bottom": "0px",
        "left": "0px",
        "msBorderRadius": "2px",
        "position": "absolute",
        "right": "0px",
        "top": "0px",
      }
    }
  >
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "OBorderRadius": "2px",
          "WebkitBorderRadius": "2px",
          "borderRadius": "2px",
          "bottom": "0px",
          "left": "0px",
          "msBorderRadius": "2px",
          "overflow": "hidden",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <div
        style={
          Object {
            "MozBorderRadius": "",
            "MozBoxShadow": "",
            "OBorderRadius": "",
            "OBoxShadow": "",
            "WebkitBorderRadius": "",
            "WebkitBoxShadow": "",
            "background": "url(data:image/png;base64,00) center left",
            "borderRadius": "",
            "bottom": "0px",
            "boxShadow": "",
            "left": "0px",
            "msBorderRadius": "",
            "msBoxShadow": "",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "MozBoxShadow": undefined,
          "OBorderRadius": "2px",
          "OBoxShadow": undefined,
          "WebkitBorderRadius": "2px",
          "WebkitBoxShadow": undefined,
          "background": "linear-gradient(to bottom, rgba(34,25,77, 0) 0%,
             rgba(34,25,77, 1) 100%)",
          "borderRadius": "2px",
          "bottom": "0px",
          "boxShadow": undefined,
          "left": "0px",
          "msBorderRadius": "2px",
          "msBoxShadow": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "height": "100%",
          "margin": "0 3px",
          "outline": "none",
          "position": "relative",
        }
      }
      tabIndex={0}
    >
      <div
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "50%",
              "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "MozTransform": "translate(-2px, -9px)",
              "OBorderRadius": "50%",
              "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "OTransform": "translate(-2px, -9px)",
              "WebkitBorderRadius": "50%",
              "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "WebkitTransform": "translate(-2px, -9px)",
              "backgroundColor": "rgb(248, 248, 248)",
              "borderRadius": "50%",
              "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "height": "18px",
              "msBorderRadius": "50%",
              "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "msTransform": "translate(-2px, -9px)",
              "transform": "translate(-2px, -9px)",
              "width": "18px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AlphaPointer renders correctly 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": "50%",
      "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "MozTransform": "translate(-9px, -1px)",
      "OBorderRadius": "50%",
      "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "OTransform": "translate(-9px, -1px)",
      "WebkitBorderRadius": "50%",
      "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "WebkitTransform": "translate(-9px, -1px)",
      "backgroundColor": "rgb(248, 248, 248)",
      "borderRadius": "50%",
      "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "height": "18px",
      "msBorderRadius": "50%",
      "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "msTransform": "translate(-9px, -1px)",
      "transform": "translate(-9px, -1px)",
      "width": "18px",
    }
  }
/>
`;
